backend/python/workflows.py

```diff
@@ -0,0 +1,105 @@
+import asyncio
+from datetime import timedelta, datetime
+
+from temporalio import workflow
+from temporalio.common import RetryPolicy
+from temporalio.exceptions import ApplicationError
+
+from activities import FoodDeliveryActivities
+
+from shared_objects import OrderStatus, OrderStates, Product
+
+
+@workflow.defn(name="order")
+class OrderWorkflow:
+    def __init__(self) -> None:
+        self.order_status = None
+        self.retry_policy = RetryPolicy(maximum_interval=timedelta(seconds=5))
+
+    @workflow.run
+    async def order(self, product_id: int) -> str:
+
+        product = await workflow.execute_activity_method(
+            FoodDeliveryActivities.get_product,
+            product_id,
+            start_to_close_timeout=timedelta(seconds=10),
+            retry_policy=self.retry_policy
+        )
+
+        self.order_status = OrderStatus(product_id, OrderStates.CHARGING_CARD, None)
+        try:
+            await workflow.execute_activity_method(
+                FoodDeliveryActivities.charge_customer,
+                product,
+                start_to_close_timeout=timedelta(seconds=10),
+                retry_policy=self.retry_policy
+            )
+        except ApplicationError as e:
+            self.order_status.state = OrderStates.FAILED
+            message = "Failed to charge customer for " + product.name + ". Error: " + e.message
+            await workflow.execute_activity_method(
+                FoodDeliveryActivities.send_push_notification,
+                message,
+                start_to_close_timeout=timedelta(seconds=10),
+                retry_policy=self.retry_policy
+            )
+            raise ApplicationError(message, type=e.type)
+        self.order_status.state = OrderStates.PAID
+
```

Comment:
You are missing the await/timeout condition for the "pickedUp" signal.  Look back at the Java or .NET implementations to see where this fits in the flow